src/main/java/org/example/springboot25/service/CatPhotoService.java

```
@@ -0,0 +1,51 @@
+package org.example.springboot25.service;
+
+import org.example.springboot25.entities.CatPhoto;
+import org.example.springboot25.repositories.CatPhotoRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class CatPhotoService {
+
+    private final CatPhotoRepository catPhotoRepository;
+
+    public CatPhotoService(CatPhotoRepository catPhotoRepository) {
+        this.catPhotoRepository = catPhotoRepository;
+    }
+
+    // 🔹 Get all Photos
+    public List<CatPhoto> getAllCatPhotos() {
+        return catPhotoRepository.findAll();
+    }
+
+    // 🔹 Get all Photos based on ID
+    public Optional<CatPhoto> getCatPhotoById(Long id) {
+        return catPhotoRepository.findById(id);
+    }
```

You could throw a NotFoundException here if no photo is found, instead of returning Optional.